mmdet3d/core/bbox/coders/monoflex_bbox_coder.py

```diff
@@ -0,0 +1,366 @@
+                                          1] - pred_keypoints[:, [4, 6], 1]
+        corner_13_height = pred_keypoints[:, [1, 3],
+                                          1] - pred_keypoints[:, [5, 7], 1]
+
+        pred_keypoint_depths = {'center': [], 'corner_02': [], 'corner_13': []}
+
+        for idx, gt_idx in enumerate(
+                torch.unique(batch_idxs, sorted=True).tolist()):
+            calib = calibs[idx]
+            corr_pts_idx = torch.nonzero(batch_idxs == gt_idx).squeeze(-1)
+            center_depth = calib.f_u * pred_height_3D[corr_pts_idx] / (
+                F.relu(center_height[corr_pts_idx]) * self.down_ratio +
+                self.EPS)
+            corner_02_depth = calib.f_u * pred_height_3D[
+                corr_pts_idx].unsqueeze(-1) / (
+                    F.relu(corner_02_height[corr_pts_idx]) * self.down_ratio +
+                    self.EPS)
+            corner_13_depth = calib.f_u * pred_height_3D[
+                corr_pts_idx].unsqueeze(-1) / (
+                    F.relu(corner_13_height[corr_pts_idx]) * self.down_ratio +
+                    self.EPS)
+
+            corner_02_depth = corner_02_depth.mean(dim=1)
+            corner_13_depth = corner_13_depth.mean(dim=1)
+
+            pred_keypoint_depths['center'].append(center_depth)
+            pred_keypoint_depths['corner_02'].append(corner_02_depth)
+            pred_keypoint_depths['corner_13'].append(corner_13_depth)
+
+        for key, depths in pred_keypoint_depths.items():
+            pred_keypoint_depths[key] = torch.clamp(
+                torch.cat(depths),
+                min=self.depth_range[0],
+                max=self.depth_range[1])
+
+        pred_depths = torch.stack(
+            [depth for depth in pred_keypoint_depths.values()], dim=1)
+
+        return pred_depths
+
+    def decode_dimension(self, cls_id, dims_offset):
+        '''retrieve object dimensions
+        Args:
+                cls_id: each object id
+                dims_offset: dimension offsets, shape = (N, 3)
+
+        Returns:
+
+        '''
+        cls_id = cls_id.flatten().long()
+        cls_dimension_mean = self.dim_mean[cls_id, :]
+
+        if self.dim_modes[0] == 'exp':
+            dims_offset = dims_offset.exp()
+
+        if self.dim_modes[2]:
+            cls_dimension_std = self.dim_std[cls_id, :]
+            dimensions = dims_offset * cls_dimension_std + cls_dimension_mean
+        else:
+            dimensions = dims_offset * cls_dimension_mean
+
+        return dimensions
+
+    def decode_axes_orientation(self, vector_ori, locations):
+        '''retrieve object orientation
+        Args:
+                vector_ori: local orientation in
+                    [axis_cls, head_cls, sin, cos] format.
+                locations: object location.
+
+        Returns: for training we only need roty
+                            for testing we need both alpha and roty
+
+        '''
+        if self.multibin:
```

Comment:
It's strange because I have not found any variable named `multibin`  in this class or its superclass.